### PR TITLE
chore: allow singup based on auth methods

### PR DESCRIPTION
**Pull request by JeevaRamu0104**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
image

## Checklist

- [ ] I ran `npm run re:build`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/context/AuthInfoProvider.res [hunk 1]
```diff
@@ -4,11 +4,17 @@ type defaultProviderTypes = {
...
+    array<SSOTypes.authMethodResponseType> => array<SSOTypes.authMethodResponseType>
+  ) => unit,
+  authMethods: array<SSOTypes.authMethodResponseType>,
 }
 let defaultContextValue = {
   authStatus: CheckingAuthStatus,
   setAuthStatus: _ => (),
   setAuthStateToLogout: _ => (),
+  setAuthMethods: _ => (),
+  authMethods: [],
```
Comment (Riddhiagrawal001) [inline src/context/AuthInfoProvider.res:17]: Can we put default as `AuthUtils.defaultListOfAuth`

File: src/entryPoints/AuthModule/AuthModuleHooks.res [hunk 2]
```diff
@@ -13,14 +33,157 @@ let useAuthMethods = () => {
...
+    let method: SSOTypes.authMethodTypes = PASSWORD
+    featureFlagValues.email && getAuthMethod(method)->Option.isSome
+  }, [authMethods])
+
+  let isSignUpAllowed = React.useCallback1(() => {
+    open SSOTypes
+    let magicLinkmethod = getAuthMethod(MAGIC_LINK)
+    let passwordmethod = getAuthMethod(PASSWORD)
+
+    let isSingUpAllowdinMagicLink = switch magicLinkmethod {
```
Comment (Riddhiagrawal001) [inline src/entryPoints/AuthModule/AuthModuleHooks.res:89]: typo `isSignUpAllowedinMagicLink`

File: src/entryPoints/AuthModule/AuthModuleHooks.res [hunk 3]
```diff
@@ -13,14 +33,157 @@ let useAuthMethods = () => {
...
+    | Some(magicLinkData) => magicLinkData->Array.some(v => v.allow_signup)
+    | None => false
     }
+    let isSingUpAllowdinPassword = switch passwordmethod {
```
Comment (Riddhiagrawal001) [inline src/entryPoints/AuthModule/AuthModuleHooks.res:93]: typo `isSignUpAllowedinPassword`

File: src/entryPoints/AuthModule/AuthModuleHooks.res [hunk 4]
```diff
@@ -13,14 +33,157 @@ let useAuthMethods = () => {
...
+    } else if isSingUpAllowdinPassword {
+      (true, PASSWORD)
+    } else if !isTotpFeatureDisable && emailFeatureFlagEnable {
+      (true, MAGIC_LINK)
+    } else if !isTotpFeatureDisable {
+      (true, PASSWORD)
+    } else {
+      (false, INVALID)
+    }
+  }, [authMethods])
```
Comment (Riddhiagrawal001) [inline src/entryPoints/AuthModule/AuthModuleHooks.res:115]: Can we write a comment here what is the tuple response

File: src/entryPoints/AuthModule/AuthUtils.res [hunk 5]
```diff
@@ -64,11 +64,20 @@ let getUserInfoDetailsFromLocalStorage = () => {
...
     auth_method: {
       \"type": PASSWORD,
-      name: #Email_Password,
+      name: #Password,
+    },
+    allow_signup: true,
+  },
+  {
+    id: "defaultmagicLinkId",
+    auth_id: "defaulmagicLinkId",
```
Comment (Riddhiagrawal001) [inline src/entryPoints/AuthModule/AuthUtils.res:77]: typo `defaultmagicLinkId`

File: src/entryPoints/AuthModule/AuthUtils.res [hunk 6]
```diff
@@ -64,11 +64,20 @@ let getUserInfoDetailsFromLocalStorage = () => {
...
+    auth_method: {
+      \"type": MAGIC_LINK,
+      name: #Magic_Link,
     },
     allow_signup: true,
```
Comment (Riddhiagrawal001) [inline src/entryPoints/AuthModule/AuthUtils.res:82]: Does Magic_Link allows signup in default case?

File: src/entryPoints/AuthModule/AuthWrapper.res [hunk 7]
```diff
@@ -164,18 +150,23 @@ let make = (~children) => {
...
+          <UIUtils.RenderIf condition={checkAuthMethodExists([OPEN_ID_CONNECT])}>
+            {PreLoginUtils.divider}
+          </UIUtils.RenderIf>
+          {authMethods
+          ->Array.mapWithIndex((authMethod, index) =>
+            <React.Fragment key={index->Int.toString}>
+              {authMethod->renderComponentForAuthTypes}
+            </React.Fragment>
+          )
+          ->React.array}
```
Comment (Riddhiagrawal001) [inline src/entryPoints/AuthModule/AuthWrapper.res:167]:
```suggestion
          <UIUtils.RenderIf condition={checkAuthMethodExists([OPEN_ID_CONNECT])}>
            {PreLoginUtils.divider}
             {authMethods
          ->Array.mapWithIndex((authMethod, index) =>
            <React.Fragment key={index->Int.toString}>
              {authMethod->renderComponentForAuthTypes}
            </React.Fragment>
          )
          ->React.array}
          </UIUtils.RenderIf>
         
```
As the component is only called whenever there is open_id_connect

File: src/entryPoints/AuthModule/AuthUtils.res (hunk 6 above)
Comment (JeevaRamu0104) [reply src/entryPoints/AuthModule/AuthUtils.res:82]: yes

**Review by Riddhiagrawal001 — APPROVED** [review]

**Pull request merged**